Fix typos in mission dialogue

ThrawnCA opened the pull request:
-----------------------
**Bugfix:**

## Fix Details
Typo fixes

- Restore missing double quotes at end of dialogue, and remove some surplus ones.
- Fix some spelling and grammar errors.

## Testing Done
None.

Amazinite commented:
You'll need to update the [codespell exclude file](https://github.com/endless-sky/endless-sky/blob/master/.codespell.exclude) for the one line that you changed that is not erroring on "vill."
Also, please refrain from eating the whitespace.

ThrawnCA commented:
> Also, please refrain from eating the whitespace.

Sorry, I had Geany set to clean up whitespace by default. I've switched it off for this project now (and already had to go through reverting its attempts to replace tabs with spaces). I figured trimming a few trailing spaces was harmless, though. But that rule is now switched off for this project too.

EjoThims commented on data/hai/hai reveal 4 middle.txt:
```diff
@@ -372,7 +372,7 @@ mission "Hai Reveal: Pirate Troubles [1]"
 					goto explain
 				`	"What issue?"`
-			`	"I understand that you just defended this system from human pirates. That is the issue that needs resolved.`
+			`	"I understand that you just defended this system from human pirates. That is the issue that needs resolving.`
```
```suggestion
			`	"I understand that you just defended this system from human pirates. That is the issue that needs to be resolved.`
```

This isn't necessarily incorrect (especially as spoken dialogue), but this also isn't the simplest fix.

Suggesting the change to a more fitting fix, but dropping "to be" is such a very common convention in speech that I really think this one may be best left as is.

ThrawnCA replied:
I understand that "to be" is often dropped, but "needs resolved" is definitely not valid. "Needs to be resolved" would be fine.

EjoThims replied:
Spoken dialogue not being grammatically correct is wholly valid, as long as it is intended. ;)

EjoThims commented on data/hai/hai reveal 5 philosophers.txt:
```diff
@@ -850,7 +850,7 @@ mission "Hai Reveal [B09] Scanning Devil-Hide"
 			action
 				set "Hai Reveal [B09] Scanning Devil-Hide: label moon"
-			`	The moon? Hmmm... okay, it's just an ordinary chunk of rock, but radiation from the planet is disturbing the sensors. I can only see a few dozen meters below ground, even though the scanner should be able to look down at least ten times that far... Aha!"`
+			`	"The moon? Hmmm... okay, it's just an ordinary chunk of rock, but radiation from the planet is disturbing the sensors. I can only see a few dozen meters below ground, even though the scanner should be able to look down at least ten times that far... Aha!`
```
Like 846 above could use its trailing quotes clipped the same as this one, actually. :)

EjoThims reviewed (changes requested):
Lots of good catches, just those couple of notes.

MasterOfGrey reviewed (approved):
I’m quite impressed, and more than a little surprised, that you found that many. They all look good to me though.
Some of the white space removals in the middle of missions are unnecessary, but otherwise. Nicely done!

ThrawnCA commented:
> I’m quite impressed, and more than a little surprised, that you found that many.

The missing double quote in https://github.com/endless-sky/endless-sky/pull/7940 inspired me to do a search for similar issues, so I came up with a regex:

    grep -r '^[^"]*"[^"]*$' data

to identify lines containing just one double quote mark.

And I noticed a handful of spelling errors while reviewing the results.

(Ideally it would search for any odd number of double quotes, though that would be more complex.)

> Nicely done!

Thanks :)

tibetiroka reviewed (approved):
The fixes are good, but I'm also not sure about the whitespace changes. I guess they can stay.

Zitchas reviewed (approved):
I haven't had the chance to review everything, but I have looked over the changes in the Remnant files, and they look good.

Amazinite commented on data/human/deep missions.txt:
```diff
@@ -1292,11 +1292,11 @@ mission "Deep: Remnant 0"
 		has "First Contact: Hai: offered"
 		has "main plot completed"
-	
+
```
```suggestion
	
```

Amazinite commented on data/human/deep missions.txt:
```diff
@@ -2428,15 +2428,15 @@ mission "Deep: Scientist Rescue 3B"
 				`	"Thank you so much! I'll make sure that this gets put to good use."`
 				`	"Really? You shouldn't have."`
-			
+
```
```suggestion
			
```

Amazinite commented on data/remnant/remnant 1 introduction.txt:
```diff
@@ -1373,12 +1373,12 @@ mission "Remnant: Deep Surveillance"
+			`	Ivan then takes a seat in his own cushioned chair. "Welcome to my office. This is miles above what I'm used to. After gaining the attention of the Deep government with the discoveries we were making, they immediately had my office moved from the dingy complex it used to be at to here. I have you to thank for much of that work.`
 				goto reason
-			
+
```
```suggestion
			
```

Amazinite commented on data/remnant/remnant 1 introduction.txt:
```diff
@@ -1373,12 +1373,12 @@ mission "Remnant: Deep Surveillance"
 			label second
 			`	Ivan then takes a seat in his own cushioned chair. "Welcome to my office. Much better than the old place, isn't it? After gaining the attention of the Deep government with the discoveries we were making, they immediately had my office moved from that dingy complex to here. I have you to thank for much of that work.`
-			
+
```
```suggestion
			
```

Amazinite commented on data/sheragi/archaeology missions.txt:
```diff
@@ -648,16 +648,16 @@ mission "Sheragi Archaeology: The Box 2"
 	source "Zug"
 	destination "Vinci"
-	
+
```
```suggestion
	
```

Amazinite commented on data/sheragi/archaeology missions.txt:
```diff
@@ -648,16 +648,16 @@ mission "Sheragi Archaeology: The Box 2"
 	to offer
 		has "Sheragi Archaeology: The Box 1: done"
-	
+
```
```suggestion
	
```

Amazinite commented on data/sheragi/archaeology missions.txt:
```diff
@@ -875,16 +875,16 @@ mission "Sheragi Archaeology: The Box 4"
+			`	"Come back to me when you have it, then, and I shall provide you with the information. We have all the time in the universe. Until you die, or someone decides to destroy the cube, that is."`
 				goto exit
-			
+
```
```suggestion
			
```

Amazinite commented on data/sheragi/archaeology missions.txt:
```diff
@@ -875,16 +875,16 @@ mission "Sheragi Archaeology: The Box 4"
 			label time
 			`	"Come back when you're ready, then. We have all the time in the universe. Until you die, or someone decides to destroy the cube, that is. It's all up to you."`
-			
+
```
```suggestion
			
```

Amazinite commented on data/sheragi/archaeology missions.txt:
```diff
@@ -875,16 +875,16 @@ mission "Sheragi Archaeology: The Box 4"
 			`	They gesture, and the men promptly and diligently escort you back out to your ship.`
 				defer
-			
+
```
```suggestion
			
```

Amazinite commented on data/sheragi/archaeology missions.txt:
```diff
@@ -1417,27 +1417,27 @@ mission "Sheragi Archaeology: The Emerald Sword 3"
 			choice
 				`	"What is it?"`
-			
+
```
```suggestion
			
```

Amazinite commented on data/sheragi/archaeology missions.txt:
```diff
@@ -1417,27 +1417,27 @@ mission "Sheragi Archaeology: The Emerald Sword 3"
 				`	"Does that mean this was a daycare of sorts?"`
 					goto daycare
-			
+
```
```suggestion
			
```

Amazinite commented on data/sheragi/archaeology missions.txt:
```diff
@@ -1417,27 +1417,27 @@ mission "Sheragi Archaeology: The Emerald Sword 3"
 			`	"Yes. At least one, maybe two generations of Sheragi could have been born in that time."`
 				goto walls
-				
+
```
```suggestion
			
```

Amazinite commented on data/sheragi/archaeology missions.txt:
```diff
@@ -1417,27 +1417,27 @@ mission "Sheragi Archaeology: The Emerald Sword 3"
 			label daycare
 			`	Eyama walks over to one of the objects on the ground and picks it up. "These are toys. So yes. This does seem to have been converted into a daycare."`
-			
+
```
```suggestion
			
```
